.github/workflows/build.yml

```diff
@@ -0,0 +1,90 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Release with mvn
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - 
+      name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: maven
+    - 
+      name: app version
+      run: |
+        echo "app_ver=$(mvn -B help:evaluate -Dexpression=project.version -q -DforceStdout)"
+        echo "app_ver=$(mvn -B help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV
+    -
+      name: Build with Maven
+      run: mvn -B clean integration-test package assembly:single docker:build
+
+# Docker hub action
+    -
+      name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }} 
+    -
+      name: push snapshot version
+      run: |
+        echo publishing docker container $app_ver
+        docker push docker.io/obsidiandynamics/kafdrop:$app_ver
```

Comment:
Couldn't this accidentally overwrite an existing Dockerhub version if one forgets to add "-SNAPSHOT" after a release commit? Afaik this push with a new image id would succeed, while the Github release artifact wouln't change (see line 84).

Maybe it's better to first upload the Github release artifact and only if that succeeds, upload to Dockerhub and Bintray?

Reply:
I like the idea. @mcs  I updated the PR to the order: build, github release, bintray and docker hub.